pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -336,6 +336,27 @@ static Type promoteTypes(ConversionPatternRewriter &rewriter, Location loc,
   return bestType;
 }
 
+struct NegIOp {
+  Value create(ConversionPatternRewriter &rewriter, Location loc,
+               Type resultType, ArrayRef<Value> operands,
+               ArrayRef<Type> types) {
+    auto zero = rewriter.create<mlir::ConstantIntOp>(loc, 0, resultType);
+    auto neg = rewriter.create<mlir::SubIOp>(loc, zero, operands[0]);
+    return neg.getResult();
+  }
+};
+
+struct NotOp {
+  Value create(ConversionPatternRewriter &rewriter, Location loc,
+               Type resultType, ArrayRef<Value> operands,
+               ArrayRef<Type> types) {
+    // -(x + 1) = -1 - x
+    auto negOne = rewriter.create<mlir::ConstantIntOp>(loc, -1, resultType);
+    auto sub = rewriter.create<mlir::SubIOp>(loc, negOne, operands[0]);
+    return sub.getResult();
```

Comment:
We know the signed ints will always be represented with 2's complement?

I guess there would be ambiguity around signedness with the int ops otherwise...

Reply:
AFAIK all machines seem to use 2's complement. I think LLVM is based on that assumption. Also, there doesn't seem to be a `std.negi` or ` std.bit_not`. It looks like LLVM does have intrinsics that we could plumb thru at some point.